### ISSUE TITLE
[⚡ Feature]: Add support for other methods in the exported worker

### ISSUE THREAD

**Issue by mmiszy** [cloudflare/next-on-pages#477]
### Description

Currently, the worker template only implements the `fetch` method.
Would it be possible to add other methods as well?

- `tail`
- `trace`
- `scheduled`
- `test`
- `email`
- `queue`

### Would you like to help?

- [X] Would you like to help implement this feature?

**Comment by dario-piotrowicz** [comment]
Thanks a lot for the issue @mmiszy, unfortunately the methods you shared are only currently available on Workers, and not Pages, so it is not something that we can add to next-on-pages (since we do create a Worker but that is used in a Pages application, it is not deployed as a simple Worker)

We'll definitely look into implementing those methods when/if it will be possible to do so (especially the `scheduled` one as we'd need it to match the Vercel one) but currently there isn't really a way to implement them sorry 😢 

The only viable alternative is I believe to create separate workers which do implement such methods and from them you can then send requests to your next-on-pages application, sorry to disappoint you 😓

**Comment by mmiszy** [comment]
I've implemented a separate worker for now, thanks!

**Comment by dario-piotrowicz** [comment]
@mmiszy yeah that's the way to go for now, hopefully it's not too inconvenient for your use case 🙂 

I've marked the issue as `blocked by external` as we might likely be able to support the other methods/triggers at some point so we might keep this issue around as a reminder 👍

**Comment by dario-piotrowicz** [comment]
Hello 🙂
Earlier this year we released our [Cloudflare adapter for OpenNext](https://opennext.js.org/cloudflare), and we have been continuously investing effort in its improvement since then. Existing deployments can continue to use the next-on-pages tooling, but the Cloudflare adapter for OpenNext is currently stable and recommended for deployment of Next.js applications to Cloudflare.

As such, we will no longer be maintaining next-on-pages. We recently deprecated the package on npm and we plan to archive this repo on Monday Sep 29, 2025. In preparation for archival, we are closing all open issues and PRs.

If you have concerns, please feel free to reply here before the repo is archived (after that, it will no longer be possible to comment) or to reach out to our team on [Discord](https://discord.com/invite/cloudflaredev).

Thanks so much for using and contributing to next-on-pages 🧡
